curator/src/util.py

```diff
@@ -3,16 +3,21 @@
 import os
 import sys
 import logging
+import urllib2
+import ssl
+import https
 
 def create_logger(name):
-    logger = logging.getLogger(name)
-    lvl = logging._levelNames.get(os.getenv('CURATOR_SCRIPT_LOG_LEVEL', 'INFO'), None)
-    if not lvl:
-        print >> sys.stderr, 'The CURATOR_SCRIPT_LOG_LEVEL must be one of CRITICAL, ERROR, WARNING, INFO or DEBUG'
-        return None
-    logger.setLevel(lvl)
-    lh = logging.StreamHandler()
-    lh.setLevel(lvl)
-    lh.setFormatter(logging._defaultFormatter)
-    logger.addHandler(lh)
-    return logger
+    lvl = logging._levelNames.get(os.getenv('CURATOR_SCRIPT_LOG_LEVEL', 'INFO'))
+    logging.basicConfig(level=lvl)
+    return logging.getLogger(name)
+
+def create_auth_opener(key, cert, ca):
+    handlers = []
+    handlers.append( https.HTTPSClientAuthHandler(
+        key = key,
+        cert = cert,
+        ca_certs = ca,
+        ssl_version = ssl.PROTOCOL_SSLv23,
+        ciphers = 'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256' ) )
```

Comment:
is it necessary to provide `ssl_version` and `ciphers`?  If so - will have to figure out how to get the "best" default values from the library - I don't think we should hardcode these here - @tiran ?

Reply:
I advise against hard-coding cipher strings. This code will break as soon as TLS 1.3 is available. ``PROTOCOL_SSLv23`` will negotiate the highest available TLS version, which may be TLS 1.3. But ``TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256`` is not a valid TLS 1.3 cipher suite.

In fact ``TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256`` is not a valid OpenSSL cipher suite at all.

Reply:
thanks @tiran - so `ssl_version = ssl.PROTOCOL_SSLv23` is ok - can we just omit `ciphers`?

Reply:
@richm The code has security bugs. I left some comments.